src/settings.ts

```diff
@@ -19,10 +20,13 @@ export class Settings {
                     && updatedConfig.syncWithFolderExplorer)) {
                 this._dependencyConfig = updatedConfig;
                 commands.executeCommand(Commands.VIEW_PACKAGE_REFRESH);
+            } else if (updatedConfig.autoRefresh !== this._dependencyConfig.autoRefresh) {
+                SyncHandler.updateFileWatcher(Settings.autoRefresh());
             } else {
                 this._dependencyConfig = updatedConfig;
```

Comment:
Move this line to the outer, make sure the cached config is synced with the latest.